src/main/java/zmq/io/StreamEngine.java

```diff
@@ -254,14 +254,18 @@ public void plug(IOThread ioThread, SessionBase session)
         handle = ioObject.addFd(fd);
         ioError = false;
 
+        //  Make sure batch sizes match large buffer sizes
+        final int inBatchSize = options.rcvbuf > Config.IN_BATCH_SIZE.getValue() ? options.rcvbuf : Config.IN_BATCH_SIZE.getValue();
+        final int outBatchSize = options.sndbuf > Config.OUT_BATCH_SIZE.getValue() ? options.sndbuf : Config.OUT_BATCH_SIZE.getValue();
```

Comment:
These could be simplified using Math.max:

```java
final int inBatchSize = Math.max(options.rcvbuf, Config.IN_BATCH_SIZE.getValue());
final int outBatchSize = Math.max(options.sndbuf, Config.OUT_BATCH_SIZE.getValue());
```

Reply:
Seems reasonable.